File: frontend/src/types/customer.ts
export interface Customer {
  id: number
  name: string
  email: string | null
  phone: string | null
  company: string | null
  address: string | null
  contract_count: number
  created_at: string
  updated_at: string
}

export interface CustomerCreate {
  name: string
  email?: string | null
  phone?: string | null
  company?: string | null
  address?: string | null
}

export type CustomerUpdate = Partial<CustomerCreate>

File: backend/app/schemas/customer.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, EmailStr


class CustomerBase(BaseModel):
    name: str
    email: EmailStr | None = None
    phone: str | None = None
    company: str | None = None
    address: str | None = None


class CustomerCreate(CustomerBase):
    pass


class CustomerUpdate(BaseModel):
    name: str | None = None
    email: EmailStr | None = None
    phone: str | None = None
    company: str | None = None
    address: str | None = None


class CustomerResponse(CustomerBase):
    id: int
    contract_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
